Library/Infrastructure/NfieldConnection.cs

```diff
@@ -67,27 +72,36 @@ public Task<bool> SignInAsync(string domainName, string username, string passwor
         {
             if (Client == null)
             {
-                Client = (INfieldHttpClient)DependencyResolver.Current.Resolve(typeof(INfieldHttpClient));
+                Client = new DefaultNfieldHttpClient(_httpClient);
             }
+
             var data = new Dictionary<string, string>
                 {
                     {"Domain", domainName},
                     {"Username", username},
                     {"Password", password}
                 };
             var content = new FormUrlEncodedContent(data);
-            return Client.PostAsync(NfieldServerUri + "SignIn", content)
+
+            // client will update the Token
+            return Client.PostAsync(new Uri(NfieldServerUri, "SignIn"), content)
                 .ContinueWith(responseMessageTask =>
                 {
                     var result = responseMessageTask.Result;
-
-                    _token = Client.AuthToken;
-                    
                     return result.StatusCode == HttpStatusCode.OK;
-
                 }).FlattenExceptions();
         }
 
+        public Task SignInAsync(string domainName, Func<Task<string>> provideTokenAsync)
```

Comment:
There's no comment. Why doesn't this method do anything?

Reply:
The comment on the interface is helpful, but it says "sign in" but it doesn't appear to actually do that.

Reply:
We were struggling to find a better name :smile: if you have a suggestion we can change it.

Reply:
How about `RegisterTokenProvider`? It also doesn't need to be async.